packages/core/src/stream-loading/state-manipulator.ts

```diff
@@ -36,6 +36,10 @@ export interface ApplyLogToStateOpts {
   throwOnConflict: boolean
 }
 
+export interface ResetStateToCommitOpts {
```

Comment:
Nitpick, as a way to share an idea. Kind of suggestion to think about. Not a critique in any way, as it is half-baked. Feel free to disregard.

TS provides `type` and `interface` constructs. I find it easier to reason about them, if we use `type` for things like types of parameters or shapes of plain JS objects. It is a standalone entity. Inheritance is out of the equation for `type`. If something is declared as an `interface` though, it conveys a meaning that it has to be implemented by some class. That is, if you refactor an interface, beware of the dragons downstream. Would be curious to hear your thoughts about that heuristic someday.

Reply:
Yeah I've never felt clear on when to use `interface` vs `type`.  That is helpful context for thinking about it.  I'll do a separate follow-up PR to switch all of these over to `type`